### PR TITLE
Skip invisible path head of initial walk

### PR DESCRIPTION
**Pull request by Alberth289346**
A step towards getting the first patient quicker (issue #649 ) in the hospital.

As discussed by @mugmuggy at https://github.com/CorsixTH/CorsixTH/issues/649#issuecomment-356282253 original game spawns closer to the hospital. This patch does that.

Patch is more readable when viewing commits rather than the entire branch.

The impact of this patch seems small judging by a first test-run, so the problem is also in the patient spawning process.

Addresses #649

### CODE REVIEW

File: CorsixTH/Lua/humanoid_actions/seek_reception.lua [hunk 1]
```diff
@@ -23,20 +23,126 @@ class "SeekReceptionAction" (HumanoidAction)
...
 
-function SeekReceptionAction:SeekReceptionAction()
+-- Number of tile traversals to keep before reaching a hospital.
+local min_street_length = 32
+
+--! Seek the reception of the hospital for registration.
+--!param trim_head If true, trim the first part of the path to the hospital.
+function SeekReceptionAction:SeekReceptionAction(trim_head)
   self:HumanoidAction("seek_reception")
+  self.trim_head = not not trim_head
```
[the remark this answers is not in the document]
Comment (Alberth289346) [reply CorsixTH/Lua/humanoid_actions/seek_reception.lua:36]: "not not x" is a way to normalize x to true/false, so `nil` becomes `false` as you would expect.

**Comment by Alberth289346** [comment]
> Have you checked that an existing save with patients already seeking reception don't get bugged with this?

I haven't but the path gets trimmed before starting to walk so that should be fine. Also, seek actions that exist but have not started have no `trim_head` and thus won't apply trimming the start.

File: CorsixTH/Lua/humanoid_actions/seek_reception.lua [hunk 2]
```diff
@@ -23,20 +23,126 @@ class "SeekReceptionAction" (HumanoidAction)
...
+  if tile_top_x + tile_size[1] / 2 < 0 then return false end
+  return tile_top_x - tile_size[1] / 2 < scr_size[1]
+end
+
+--! Remove some tiles from the start of the path with tiles that are not
+--  visible at the screen.
+--!param humanoid (Humanoid) Humanoid walking the path
+--!param path_x (array int) X coordinates of the path, modified in-place.
+--!param path_y (array int) Y coordinates of the path, modified in-place.
+local function trimInvisibleHead(humanoid, path_x, path_y)
```
[the remark this answers is not in the document]
Comment (Alberth289346) [reply CorsixTH/Lua/humanoid_actions/seek_reception.lua:100]: They may use a bigger zoom value to see what they're doing :p

**Comment by TheCycoONE** [comment]
This PR moves us further from determinism and multiplayer by making the game depend on the configured screen resolution. I'm fairly opposed to that, and doubly so when it didn't have the intended effect.

Can we call it a proof of concept / experiment, copy forward the code cleanup and discard the change otherwise?

**Comment by lewri** [comment]
> This PR moves us further from determinism and multiplayer by making the game depend on the configured screen resolution

You could set it to be disabled for multiplayer. Plus as Albert added a parameter for the trim you could also just use that?

**Comment by TheCycoONE** [comment]
Yes, but he also said it didn't make much difference, and it makes the game play different depending on the size of your monitor, which is just bizarre. I don't see the advantage of having this. Change for change sake doesn't justify it.

**Comment by Alberth289346** [comment]
> Can we call it a proof of concept / experiment, copy forward the code cleanup and discard the change otherwise?

Sure, no problem.

**Comment by Alberth289346** [comment]
Wrong direction.

**Pull request closed without merging**